contracts/minter/src/state.rs

```diff
@@ -0,0 +1,22 @@
+use cw_utils::Expiration;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Addr, Coin, Empty};
+use cw_storage_plus::{Item, Map};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    pub admin: Addr,
+    pub base_token_uri: String,
+    pub num_tokens: u64,
+    pub sg721_code_id: u64,
+    pub unit_price: Coin,
+    pub whitelist: Option<Addr>,
+    pub start_time: Option<Expiration>,
+    pub per_address_limit: Option<u32>,
+}
+
+pub const CONFIG: Item<Config> = Item::new("config");
+pub const SG721_ADDRESS: Item<Addr> = Item::new("sg721_address");
+pub const MINTABLE_TOKEN_IDS: Map<u64, Empty> = Map::new("mt");
```

Comment:
Instead of removing data, you can set the value as bool and flag it true. In fact `Empty` uses more space than `bool`